electrum_dash/dash_tx.py

```diff
@@ -698,156 +700,97 @@ def read_vds(cls, vds):
+
+    def __init__(self, version, index, fee, signHeight, quorumHash, quorumSig):
+        self.version = version              # uint8_t
+        self.index = index                  # uint64
+        self.fee = fee                      # uint32
+        self.signHeight = signHeight        # uint32
+        self.quorumHash = quorumHash        # bytes(32)
+        self.quorumSig = quorumSig          # bytes(96)
 
     def __str__(self):
-        return ('SubTxCloseAccount Version: %s\n'
-                'regTxHash: %s\n'
-                'hashPrevSubTx: %s\n'
-                'creditFee: %s\n'
-                % (self.version,
-                   bh2u(self.regTxHash[::-1]),
-                   bh2u(self.hashPrevSubTx[::-1]),
-                   self.creditFee))
+        return ('AssetUnlockTx Version: {}\n'
+                'Index: {}\n'
+                'Fee: {}\n'
+                'Sign Height: {}\n'
+                'Quorum Hash: {}\n'
+                'Quorum Signature: {}\n'
+                .format(
+                    self.version,
+                    self.index,
+                    self.fee,
+                    self.signHeight,
+                    bh2u(self.quorumHash[::-1]),
+                    bh2u(self.quorumSig)
+                ))
 
     def serialize(self):
-        assert len(self.regTxHash) == 32, \
-            f'{len(self.regTxHash)} not 32'
-        assert len(self.hashPrevSubTx) == 32, \
-            f'{len(self.hashPrevSubTx)} not 32'
-        assert len(self.payloadSig) == 96, \
-            f'{len(self.payloadSig)} not 96'
-        return (
-            struct.pack('<H', self.version) +           # version
-            self.regTxHash +                            # regTxHash
-            self.hashPrevSubTx +                        # hashPrevSubTx
-            struct.pack('<q', self.creditFee) +         # creditFee
-            self.payloadSig                             # payloadSig
-        )
+        res = b''
+        res += struct.pack('<B', self.version)     # version (uint8_t)
+        res += struct.pack('<Q', self.index)       # index (uint64)
+        res += struct.pack('<I', self.fee)         # fee (uint32)
+        res += struct.pack('<I', self.signHeight)  # signHeight (uint32)
+        res += self.quorumHash                     # quorumHash (32 bytes)
+        res += self.quorumSig                      # quorumSig (96 bytes)
+        return res
 
     @classmethod
     def read_vds(cls, vds):
-        return DashSubTxCloseAccount(
-            vds.read_uint16(),                          # version
-            vds.read_bytes(32),                         # regTxHash
-            vds.read_bytes(32),                         # hashPrevSubTx
-            vds.read_int64(),                           # creditFee
-            vds.read_bytes(96)                          # payloadSig
-        )
+        version = read_uint8(vds)                   # uint8_t
```

Comment:
You should document field names, not the types

Reply:
fixed